src/app/shared/directives/date-highlighter.directive.ts

```diff
@@ -0,0 +1,49 @@
+import { Directive, ElementRef, Input, OnInit, Renderer2 } from '@angular/core'
+
+enum DateColors {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+  Yellow = 'yellow',
+}
+
+@Directive({
+  selector: '[appDateHighlighter]',
+})
+export class DateHighlighterDirective implements OnInit {
+  @Input() appDateHighlighter = ''
+
+  private colorClass = DateColors.Yellow
+
+  constructor(private el: ElementRef, private renderer2: Renderer2) {}
+
+  ngOnInit(): void {
+    this.defineColor()
+    this.renderer2.addClass(this.el.nativeElement, this.colorClass)
+  }
+
+  private defineColor() {
+    const date = new Date(this.appDateHighlighter).getTime() || Date.now()
+    if (date > this.getTimeFromDaysPassed(7)) {
+      this.colorClass = DateColors.Blue
+    } else if (date > this.getTimeFromMonthsPassed(1)) {
+      this.colorClass = DateColors.Green
+    } else if (date < this.getTimeFromMonthsPassed(6)) {
+      this.colorClass = DateColors.Red
+    }
+  }
+
+  private getTimeFromDaysPassed(days: number): number {
+    return new Date(Date.now() - days * 24 * 60 * 60 * 1000).getTime()
+  }
+
+  private getTimeFromMonthsPassed(monthsPassed: number) {
```

Comment:
please enable eslint `explicit-function-return-type` rule (with whatever setting you prefer),  just to have consistent return types